docs/storage/gcsupload/gcsupload_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"cloud.google.com/go/storage"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestUpload(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	client, bh, cleanupBucket := withBucketCleanup(t, tc.ProjectID, "gcs-upload-test")
+	defer cleanupBucket()
+
+	r := strings.NewReader(strings.Repeat("GCS test\n", 30))
+	wantLen := r.Len()
+
+	name := "atest.txt"
+	obj, objAttrs, err := Upload(client, r, tc.ProjectID, bh.Name, name, true)
+	if err != nil {
+		t.Fatalf("expected to successfully upload: %v", err)
+	}
+	if objAttrs == nil {
+		t.Fatal("expected back a non-nil object")
+	}
+	defer obj.Delete(context.Background())
+
+	if g, w := objAttrs.Size, int64(wantLen); g != w {
+		t.Errorf("size: got=%d want=%d", g, w)
+	}
+}
+
+func TestEnsureBucketExists(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	bucketName := "gcs-ensure-bucket-test"
```

Comment:
Yeah am gonna scrap it, I was just following in vein of listBuckets, in deed this test doesn't add anything.